utility/README.md

```diff
@@ -1,679 +1,1309 @@
+An Application that chooses to delegate trust to a Gateway will need to submit a one-time `DelegateMsg` transaction to delegate trust from the Application to the Gateway. It must include the PublicKey of the Gateway and be signed by the Application.
 
+```mermaid
+sequenceDiagram
+    actor A as Application
+    participant WS as World State
+    actor G as Gateway
 
-## 3.5 Validator Protocol
+    A ->> A: Prepare Delegate Message
+    A ->> A: Sign Request
 
-Validators are a category of actor whose responsibility is to securely process transactions and blocks using the Pocket Network 1.0 Consensus Protocol. Though most of their functional behavior is described in the external 1.0 Consensus Protocol specification and the Transaction Protocol section, a state specific Validator Protocol is needed to allow a dynamic Validator set and provide the necessary incentive layer for their operations. In order to participate in the network as a Validator, each actor is required to bond a certain amount of tokens in escrow while they are providing the Validator services. These tokens are used as collateral to disincentivize byzantine behavior. Upon registration, a Validator must inform the network of how many tokens they are staking and the public endpoint where the Validator API is securely exposed.  In addition to the required information, an optional OperatorPublicKey may be provided for non-custodial operation of the Validator. This registration message is formally known as the StakeMsg and is represented below in pseudocode:
+    A ->>+ WS: Delegate(GatewayPubKey)
+    WS ->>- A: ok
+```
 
+The following message will need to be signed by the Application's PrivateKey in order for it to be validated committed to the world state.
 
-```
-# ValidatorStakeMsg Interface
-type ValidatorStakeMsg interface {
-  GetPublicKey()  PublicKey       # The public cryptographic id of the custodial account
-  GetStakeAmount() BigInt         # The amount of uPOKT put in escrow, like a security deposit
-  GetServiceURL() ServiceURL      # The API endpoint where the validator service is provided
-  GetOperatorPubKey() PublicKey   # OPTIONAL; The non-custodial pubKey operating this node
+```go
+type DelegateMsg interface {
+  GetApplicationPublicKey() # The cryptographic ID of the Application
+  GetGatewayPublicKey() # The cryptographic ID of the Application
```

Comment:
Suggestion: If applications can delegate to multiple gateways, then there should be a mechanism to control how many “relays” a specific gateway can serve to prevent any gateway exhaustion. Imagine a scenario where “higher volumed” gateways uses all of the applications relays per session. This then could be a nuisance as other gateways would be unable to serve a request despite having N applications delegated to them. The check could be done on the servicer level as part of the relay validation and adding an allotment as part of the delegation method (with parameters of course)

Another note: A normal application could “count” the relays sent on the client layer and compare it to relaysPerSession, whereas gateways would have no idea if they reached their delegated application relay limits without parsing the servicer errors. So gateways would have to optimistically send a relay without a guaranteed success. However, with an allotment per delegation, gateways would also be able to count.